quill-core/src/main/scala/io/getquill/quotation/Parsing.scala

```diff
@@ -329,22 +331,50 @@ trait Parsing {
   private def ident(x: TermName): Ident = identClean(Ident(x.decodedName.toString))
 
   val optionOperationParser: Parser[OptionOperation] = Parser[OptionOperation] {
-    case q"$o.flatten[$t]($implicitBody)" if is[Option[Any]](o) =>
-      OptionFlatten(astParser(o))
-    case q"$o.getOrElse[$t]($body)" if is[Option[Any]](o) =>
-      OptionGetOrElse(astParser(o), astParser(body))
+
+    // For tables or specifically where the user want an unchecked map/flatMap
     case q"$o.flatMap[$t]({($alias) => $body})" if is[Option[Any]](o) =>
-      OptionFlatMap(astParser(o), identParser(alias), astParser(body))
+      if (isOptionRowType(o) || isOptionEmbedded(o))
```

Comment:
could you clarify what this condition does?

Reply:
Basically, if it's a row type (e.g. case class, tuple, or embedded), always use Unchecked since since you can't null-check these types in sql. I'll add a comment.